### PR TITLE
fix: remove guard for auth scheme for operations such as STS GetCallerIdentity to continue execution

### PR DESCRIPTION
**Pull request by ganeshnj**
## Issue \#

Fixes https://github.com/awslabs/aws-sdk-swift/issues/619

## Description of changes

With endpoint 2.0, this change was added to make sure there is auth scheme present always which wasn't the right assumption for some operation like GetCallerIdentity. This change removes this restrictions.

Companion PR https://github.com/awslabs/smithy-swift/pull/465

## New/existing dependencies impact assessment, if applicable
No new dependencies were added to this change.

## Conventional Commits

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: codegen/smithy-aws-swift-codegen/src/test/kotlin/software/amazon/smithy/aws/swift/codegen/awsquery/AWSQueryOperationStackTest.kt [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        let endpointParams = EndpointParams()
+        operation.buildStep.intercept(position: .before, middleware: EndpointResolverMiddleware<NoInputAndOutputOutputResponse, NoInputAndOutputOutputError>(endpointResolver: config.endpointResolver, endpointParams: endpointParams))
+        let apiMetadata = AWSClientRuntime.APIMetadata(serviceId: serviceName, version: "1.0.0")
+        operation.buildStep.intercept(position: .before, middleware: AWSClientRuntime.UserAgentMiddleware(metadata: AWSClientRuntime.AWSUserAgentMetadata.fromEnv(apiMetadata: apiMetadata, frameworkMetadata: config.frameworkMetadata)))
+        operation.serializeStep.intercept(position: .after, middleware: ClientRuntime.SerializableBodyMiddleware<NoInputAndOutputInput, NoInputAndOutputOutputResponse>())
+        operation.serializeStep.intercept(position: .after, middleware: ContentTypeMiddleware<NoInputAndOutputInput, NoInputAndOutputOutputResponse>(contentType: "application/x-www-form-urlencoded"))
+        operation.finalizeStep.intercept(position: .before, middleware: ClientRuntime.ContentLengthMiddleware())
```
Comment (ganeshnj) [inline codegen/smithy-aws-swift-codegen/src/test/kotlin/software/amazon/smithy/aws/swift/codegen/awsquery/AWSQueryOperationStackTest.kt:45]: this is what missing

File: codegen/smithy-aws-swift-codegen/src/main/kotlin/software/amazon/smithy/aws/swift/codegen/middleware/EndpointResolverMiddleware.kt [hunk 2]
```diff
@@ -53,14 +53,9 @@ class EndpointResolverMiddleware(
         writer.write("let endpoint = try endpointResolver.resolve(params: endpointParams)")
             .write("")
 
-        writer.openBlock("""guard let authScheme = endpoint.authScheme(name: "sigv4") else {""", "}") {
```
Comment (jbelkins) [inline codegen/smithy-aws-swift-codegen/src/main/kotlin/software/amazon/smithy/aws/swift/codegen/middleware/EndpointResolverMiddleware.kt:56]: Any idea what the rationale was for enforcing the auth scheme in the first place?

**Review by jbelkins — APPROVED** [review]
Approved, but with a question

File: codegen/smithy-aws-swift-codegen/src/main/kotlin/software/amazon/smithy/aws/swift/codegen/middleware/EndpointResolverMiddleware.kt (hunk 2 above)
Comment (ganeshnj) [reply codegen/smithy-aws-swift-codegen/src/main/kotlin/software/amazon/smithy/aws/swift/codegen/middleware/EndpointResolverMiddleware.kt:56]: That was a bad assumption that we will always a `authscheme` present.

In past also, both properties inside the authScheme were optional https://github.com/awslabs/aws-sdk-swift/blob/0.3.1/AWSClientRuntime/Sources/Middlewares/EndpointResolverMiddleware.swift#L49-L55

**Comment by jbelkins** [comment]
@ganeshnj I fired this up and `GetCallerIdentity` now works for me 🎉

**Comment by ganeshnj** [comment]
> @ganeshnj I fired this up and `GetCallerIdentity` now works for me 🎉

awesome, merging it now.

**Pull request merged**